### PR TITLE
new module: samtools/getrg

### PR DESCRIPTION
**Pull request by matthdsm**
Add module to get RG info from bam/cram files
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/samtools/getrg/main.nf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    conda (params.enable_conda ? "bioconda::samtools=1.15.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.15.1--h1170115_0' :
+        'quay.io/biocontainers/samtools:1.15.1--h1170115_0' }"
+
+    input:
+    tuple val(meta), path(input)
+
+    output:
+    tuple val(meta), file("readgroups.txt"),    emit: readgroup
```
Comment (Midnighter) [inline modules/samtools/getrg/main.nf:14]: Perhaps you can read the content directly (untested)?

```suggestion
    tuple val(meta), file("readgroups.txt").text,    emit: readgroup
```

File: modules/samtools/getrg/main.nf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
+    END_VERSIONS
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+
+    echo -e "@RG\\tID:${prefix}\\tSM:${prefix}\\tPL:ILLUMINA"
+    cat <<-END_VERSIONS > versions.yml
```
Comment (Midnighter) [inline modules/samtools/getrg/main.nf:42]: You need to create a file now

```suggestion
    echo -e "@RG\\tID:${prefix}\\tSM:${prefix}\\tPL:ILLUMINA" > readgroups.txt
    
    cat <<-END_VERSIONS > versions.yml
```

**Review by Midnighter — CHANGES_REQUESTED** [review]
Need a file in the stub now. And you could decide to parse the file content directly, I think.

File: modules/samtools/getrg/main.nf (hunk 1 above)
Comment (matthdsm) [reply modules/samtools/getrg/main.nf:14]: Oooh, didn't know that was a thing! Let me try

Comment (matthdsm) [reply modules/samtools/getrg/main.nf:14]: How would I make this compatible with pytest?

**Comment by nvnieuwk** [comment]
*ignore this:* Test comment for #2124 @ewels

**Comment by ewels** [comment]
> _ignore this:_ Test comment for #2124 @ewels

Need to break prettier first to really test it out 😉

File: modules/samtools/getrg/meta.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - input:
+      type: file
+      description: BAM/CRAM/SAM file
+      pattern: "*.{bam,cram,sam}"
```
Comment (ewels) [inline modules/samtools/getrg/meta.yml:28]: Make prettier sad
```suggestion
        type: file
        description: BAM/CRAM/SAM file
        pattern: "*.{bam,cram,sam}"
```

**Comment by ewels** [comment]
@nf-core-bot fix linting

**Comment by matthdsm** [comment]
Break my PR why don't ya 🤷🏻 😅

**Comment by ewels** [comment]
@matthdsm Fixed it again for ya 😆 

Apologies for the disruption..

File: modules/samtools/getrg/main.nf (hunk 1 above)
Comment (matthdsm) [reply modules/samtools/getrg/main.nf:14]: Doesn't seem to work a first sight, I'm getting complaints the file doesn't exist.

**Comment by matthdsm** [comment]
No problem @ewels , you can make it up by approving 😉

File: modules/samtools/getrg/main.nf (hunk 1 above)
Comment (nvnieuwk) [reply modules/samtools/getrg/main.nf:14]: The output isn't a file right? maybe this can be changed to a value in `test.yml`?

Comment (matthdsm) [reply modules/samtools/getrg/main.nf:14]: The error is from nextflow, not pytest
```console
         N E X T F L O W  ~  version 22.04.5                                                                                           
         Launching `./tests/modules/samtools/getrg/main.nf` [shrivelled_turing] DSL2 - revision: 8fef357436                            
         No such file: /Users/matdsmet/Projects/nf-core-modules/readgroups.txt                                                         
                                                                                                                                       
          -- Check script 'tests/modules/samtools/getrg/../../../../modules/samtools/getrg/main.nf' at line: 14 or see                 
         '.nextflow.log' file for more details                                                                                         
```

Comment (nvnieuwk) [reply modules/samtools/getrg/main.nf:14]: maybe this will work? 
`tuple val(meta), val(file("readgroups.txt").text.strip()),    emit: readgroup`

Comment (matthdsm) [reply modules/samtools/getrg/main.nf:14]: nope, same issue

Comment (Midnighter) [reply modules/samtools/getrg/main.nf:14]: Sorry, then you can only do it after the process 🙁

**Review by Midnighter — APPROVED** [review]

**Comment by matthdsm** [comment]
Great, thx @Midnighter

**Pull request merged**